src/test/java/com/google/cloud/tools/libraries/json/CloudLibraryTest.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.libraries.json;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+
+import com.google.common.collect.Iterables;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import org.junit.Test;
+
+/** Unit tests for {@link CloudLibrary}. */
+public final class CloudLibraryTest {
+
+  private static final String NAME = "My API";
+  private static final String ID = "myapi";
+  private static final String DOCUMENTATION = "http://documentation";
+  private static final String DESCRIPTION = "My example API";
+  private static final String TRANSPORT = "grpc";
+  private static final String ICON = "http://icon";
+  private static final String CLIENT_NAME = "My API Client";
+  private static final String CLIENT_LANGUAGE = "java";
+  private static final String CLIENT_SITE = "http://client-site";
+  private static final String CLIENT_API_REFERENCE = "http://api-reference";
+  private static final String CLIENT_INFO_TIP = "info tip";
+  private static final String CLIENT_LAUNCH_STAGE = "alpha";
+  private static final String CLIENT_SOURCE = "http://client-source";
+  private static final String CLIENT_LANGUAGE_LEVEL = "1.8.0";
+  private static final String CLIENT_MAVEN_GROUP_ID = "my-group";
+  private static final String CLIENT_MAVEN_ARTIFACT_ID = "my-artifact";
+  private static final String CLIENT_MAVEN_VERSION = "1.0.0-alpha";
+
+  @Test
+  public void parse_withFullyPopulatedJson_returnsPopulatedObject() {
```

Comment:
Since the json file is bundled into the jar and only read from there, we can use tests for the actual JSON data, not mock data. I.e. we're not trying to read any arbitrary external document but one specific document from the same repo.

Reply:
You could, but this is meant to be a lightweight unit test that has no external dependencies. Reading from the filesystem would make it much slower and it would test the integration between the model and the file. Read more at http://go/smalltest for rationale.

Reply:
The JSON is internal program source code, not an external dependency. To support the tests you're creating code paths to handle things that don't happen and making the code more general and complex than it needs to be.